libnavui-maps/src/test/java/com/mapbox/navigation/ui/maps/camera/data/ViewportDataSourceProcessorTest.kt

```diff
@@ -372,6 +372,43 @@ class ViewportDataSourceProcessorTest {
         assertArrays1(expected, actual, pointAdapter)
     }
 
+    @Test
+    fun `test getPointsToFrameOnCurrentStep, intersections disabled, cloverleaf`() {
+        val stepProgress: RouteStepProgress = mockk {
+            every { distanceTraveled } returns 0f
+            every { distanceRemaining } returns 20000000f
+            every { stepPoints } returns listOf(
+                Point.fromLngLat(20.0, 11.0),
```

Comment:
lol nice

![Screen Shot 2021-06-09 at 4 53 29 PM](https://user-images.githubusercontent.com/3021882/121443903-41458900-c943-11eb-826b-d47383f7d8ca.png)

Reply:
What are we looking at?

Reply:
It's a route that curves back behind the vehicle like a cloverleaf on-ramp. This one is taken to an extreme to prove that the unit test works. It passes when it slices the route geometry at the point where it curves more than 100º